src/plugins/intel_cpu/src/graph.cpp

```diff
@@ -1778,7 +1778,10 @@ void Graph::EnforceInferencePrecision() {
             }
 
             for (size_t i = 0; i < node->getOriginalOutputsNumber(); i++) {
-                if (node->getOriginalOutputPrecisionAtPort(i) == Precision::FP32)
+                const auto &child = node->getChildEdgesAtPort(i)[0]->getChild();
+                // exclude Convert before Range since it may cause precision loss when integter type to BF16.
+                if (!(child->getType() == Type::Range && node->getType() == Type::Convert) &&
```

Comment:
yes, re-commit this snippet.

Reply:
@dmitry-gorokhov There is no problem with "[I32] -> Convert -> [F32] -> Range". The problem only happens with "[I32] -> Convert -> [**BF16**] -> Range". For example, Convert can correctly convert from 265i to 265f, but in case of enforced BF16, it is from 265 to **264f** instead of 265f. Therefore there is a shape mismatch in some downstream nodes.